lib/engine/step/tokener.rb

```diff
@@ -29,22 +29,23 @@ def place_token(entity, city, token, teleport: false)
 
         @game.game_error('Token is already used') if token.used
 
-        token, ability = adjust_token_price_ability!(entity, token, hex)
+        token, ability = adjust_token_price_ability!(entity, token, city)
         entity.remove_ability(ability) if ability
         free = !token.price.positive?
         city.place_token(entity, token, free: free)
         unless free
           entity.spend(token.price, @game.bank)
           price_log = " for #{@game.format_currency(token.price)}"
         end
+        location_log = "(#{hex.location_name})" if hex.location_name
 
         case token.type
         when :neutral
           entity.tokens.delete(token)
           token.corporation.tokens << token
           @log << "#{entity.name} places a neutral token on #{hex.name}#{price_log}"
         else
-          @log << "#{entity.name} places a token on #{hex.name} (#{hex.location_name})#{price_log}"
+          @log << "#{entity.name} places a token on #{hex.name} #{location_log}#{price_log}"
```

Comment:
this is fine, but i think we should also add in config/game/g_1846 chicago as the location name